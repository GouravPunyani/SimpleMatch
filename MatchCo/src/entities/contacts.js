// @flow
import { observable } from 'mobx'

export class Contacts {
  @observable customerServiceEmail: ?string
  @observable feedbackEmail: ?string
  @observable mediaEmail: ?string
  @observable customerServicePhone: ?string
  @observable facebookUrl: ?string
  @observable instagramUrl: ?string
  @observable twitterUrl: ?string
  @observable companyUrl: ?string
}

export const createDefault = () => {
  const contacts = new Contacts()
  contacts.customerServiceEmail = 'made2fit@bareminerals.com'
  contacts.feedbackEmail = 'made2fit@bareminerals.com'
  contacts.mediaEmail = 'made2fit@bareminerals.com'
  contacts.customerServicePhone = '800.605.8816'
  contacts.facebookUrl = 'https://www.facebook.com/bareminerals?fref=ts'
  contacts.instagramUrl = 'https://instagram.com/bareminerals'
  contacts.twitterUrl = 'https://twitter.com/bareminerals'
  contacts.companyUrl = 'http://www.bareminerals.com'
  return contacts
}
