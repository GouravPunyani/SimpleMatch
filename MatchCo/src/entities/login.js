// @flow
import { observable, action, computed } from 'mobx'
import Validate from 'validate.js'
import I18n from 'react-native-i18n'
import { memoize, forEach, isNil, curry, keys, isEmpty } from 'ramda'

// gets the constraints, but delayed in a function due to a timing issue
// with i18n.
const getConstraints = memoize(() => {
  return {
    email: {
      presence: { message: I18n.t('errors.invalidEmail') },
      email: { message: I18n.t('errors.invalidEmail') }
    },
    password: {
      presence: { message: I18n.t('errors.passwordRequired') }
    }
  }
})

export class Login {
  @observable email: string
  @observable password: string

  @observable errors: any = {
    email: null,
    password: null
  }

  /**
   * Updates a field key with this new value.
   */
  update = action(curry(
    (key: string, value: string) => {
      this[key] = value
      this.validate(key)
    }
  ))

  @action
  async validate (...fields: string[]) {
    const currentErrors = this.errors
    const isSubset = fields.length > 0

    // grab all the errors
    const errors = Validate(this, getConstraints(), { fullMessages: false }) || {}

    // determine which fields to update
    const fieldsToUpdate = isEmpty(fields) || keys(currentErrors)

    // for subsets, reset any errors that may be error-free now
    if (isSubset && !isEmpty(errors)) {
      forEach(field => {
        if (isNil(errors[field])) currentErrors[field] = null
      }, fields)
    }

    // set each error
    forEach(field => {
      currentErrors[field] = errors[field]
    }, fieldsToUpdate)

    return isEmpty(errors)
  }

  @computed get isValid () {
    return isNil(Validate(this, getConstraints()))
  }
}
