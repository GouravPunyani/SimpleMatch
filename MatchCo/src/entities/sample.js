/**
 * A sample is a single color calibration line.
 */
export type Sample = {
  'sample_Red': number,
  'sample_Green': number,
  'sample_Blue': number,
  'sample_L': number,
  'sample_A': number,
  'sample_B': number,
  sampleLum?: number,
  specLum?: number
}
