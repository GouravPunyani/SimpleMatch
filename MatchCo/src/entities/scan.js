import type { Sample } from './sample'

/**
 * A scan is when the user performs an end-to-end scan of their face.
 */
export type Scan = {
  /** The full path to the saved scan file. */
  filename: string,
  /** The date of the scan in UTC. */
  date: Date,
  /** The device model used */
  device: string,
  /** The protocol version of the scan file. */
  version: string,
  /** The parsed sample data. */
  samples: Array<Sample>,
  /** The unparsed contents of the scan file. */
  raw: string
}
