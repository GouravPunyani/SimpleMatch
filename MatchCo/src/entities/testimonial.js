import { observable } from 'mobx'

export class Testimonial {
  @observable id: string
  @observable name: string
  @observable image: ?string
  @observable quote: string
  @observable videoUrl: ?string
}
