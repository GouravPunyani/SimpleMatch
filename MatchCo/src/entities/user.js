import { observable } from 'mobx'

/**
 * The known list of reasons for being unable to login.
 */
export type LoginErrorCode = 'wrong' | 'unknown';

export class User {
  @observable uid: ?string
  @observable email: ?string
  @observable displayName: ?string = 'Default Username' // TODO(steve): escalate
  @observable photoURL: ?string
  @observable fetching: boolean = false
  @observable errorCode: LoginErrorCode
  @observable scanTint: string = 'hotpink'
  @observable scanDate: string = 'May 32, 2017'
}
