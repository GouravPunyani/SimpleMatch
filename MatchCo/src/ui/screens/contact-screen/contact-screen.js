// @flow

import React, { Component } from 'react'
import { observer, inject } from 'mobx-react'
import { View, StyleSheet, TouchableOpacity, Dimensions } from 'react-native'
import {
  StaticLayout,
  Text,
  ContactSectionHeader,
  ContactMethod,
  ContactSocialButton
} from '../..'
import { spacing, color, fontFamily } from '../../theme'
import { ContactsStore } from '../../../stores/contacts-store'
import { sendEmail, callPhone, visitUrl } from '../../../services/opener'
import { translator } from '../../../services/translate/translator'
const { width, height } = Dimensions.get('window')

export type Props = {
  contactStore: ContactsStore,
  translator: (key: string) => string
};

@translator
@inject('contactsStore')
@observer
export class ContactScreen extends Component {
  render () {
    // grab the state out of the store
    const { translate, contactsStore } = this.props
    const { contacts } = contactsStore

    return (
      <StaticLayout>
        <View style={styles.section}>
          <ContactSectionHeader
            text={translate('contactScreen.customerServiceTitle')}
          />
          <ContactMethod
            text={contacts.customerServiceEmail}
            icon='envelope'
            onPress={() => sendEmail(contacts.customerServiceEmail)}
          />
          <ContactMethod
            text={contacts.customerServicePhone}
            icon='phone'
            onPress={() => callPhone(contacts.customerServicePhone)}
          />
          <Text>
            <Text
              style={styles.returnPolicy}
              text={translate('contactScreen.returnLabel')}
            />
            <Text text={translate('contactScreen.returnPolicy')} />
          </Text>
        </View>

        <View style={styles.section}>
          <ContactSectionHeader
            text={translate('contactScreen.featuresTitle')}
          />
          <ContactMethod
            text={contacts.feedbackEmail}
            icon='envelope'
            onPress={() => sendEmail(contacts.feedbackEmail)}
          />
          <Text text={translate('contactScreen.feedback')} />
        </View>

        <View style={styles.section}>
          <ContactSectionHeader text={translate('contactScreen.mediaTitle')} />
          <ContactMethod
            text={contacts.mediaEmail}
            icon='envelope'
            onPress={() => sendEmail(contacts.mediaEmail)}
          />
        </View>

        <View style={styles.gap} />

        <View style={styles.follow}>
          <Text
            title
            style={styles.followTitle}
            text={translate('contactScreen.followUsTitle')}
          />
          <View style={styles.social}>
            <ContactSocialButton
              network='facebook'
              onPress={() => visitUrl(contacts.facebookUrl)}
            />
            <ContactSocialButton
              network='instagram'
              onPress={() => visitUrl(contacts.instagramUrl)}
            />
            <ContactSocialButton
              network='twitter'
              onPress={() => visitUrl(contacts.twitterUrl)}
            />
          </View>
          <TouchableOpacity
            onPress={() => visitUrl(contacts.companyUrl)}
            style={styles.web}
          >
            <Text style={styles.url} text={translate('contactScreen.www')} />
          </TouchableOpacity>
        </View>
      </StaticLayout>
    )
  }
}

const styles = StyleSheet.create({
  section: { paddingBottom: spacing.large },
  follow: { alignItems: 'center' },
  followTitle: {
    color: color.brand,
    fontFamily: fontFamily.sangBlueLightItalic,
    fontSize: 18
  },
  returnPolicy: { fontFamily: fontFamily.sangBlueLightItalic },
  social: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: width*.5,
    paddingTop: spacing.medium
  },
  gap: { flex: 1 },
  web: { minHeight: 44, justifyContent: 'center' },
  url: { color: color.brand }
})
