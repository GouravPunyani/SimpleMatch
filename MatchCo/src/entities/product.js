import { observable } from 'mobx'
import { Testimonial } from './testimonial'

export class Product {
  @observable id: string
  @observable name: string
  @observable ingredients: string
  @observable description: string
  @observable price: number
  @observable sizeImperial: string
  @observable sizeMetric: string
  @observable testimonials: Array<Testimonial> = []
}
