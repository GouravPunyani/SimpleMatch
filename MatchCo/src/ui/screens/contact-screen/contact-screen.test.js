// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { createDefault } from '../../../entities/contacts'
import { ContactScreen as Screen } from './contact-screen'

test('renders correctly', () => {
  const contacts = createDefault()
  const contactsStore = { contacts }
  const tree = renderer.create(<Screen contactsStore={contactsStore} />)
  expect(tree).toMatchSnapshot()
})
