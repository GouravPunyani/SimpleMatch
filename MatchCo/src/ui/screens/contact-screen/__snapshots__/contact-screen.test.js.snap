// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
      "justifyContent": "flex-start",
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "paddingBottom": 32,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E1CD96",
          "borderBottomWidth": 1,
          "marginBottom": 2,
          "paddingBottom": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "fontFamily": "SangBleuBP-LightItalic",
              "fontSize": 16,
            },
          ]
        }
      >
        Customer Service
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "minWidth": 35,
            "paddingRight": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 23,
              "width": 35,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "color": "#E1CD96",
              "fontFamily": "NexaBold",
              "paddingLeft": 4,
            },
          ]
        }
      >
        cs@getmatch.co
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "minWidth": 35,
            "paddingRight": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 34,
              "width": 34,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "color": "#E1CD96",
              "fontFamily": "NexaBold",
              "paddingLeft": 4,
            },
          ]
        }
      >
        866.668.4300
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Nexa",
            "fontSize": 14,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "fontFamily": "SangBleuBP-LightItalic",
            },
          ]
        }
      >
        Return Policy: 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
          ]
        }
      >
        We will gladly accept returns for any reason within 30 days of delivery.
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 32,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E1CD96",
          "borderBottomWidth": 1,
          "marginBottom": 2,
          "paddingBottom": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "fontFamily": "SangBleuBP-LightItalic",
              "fontSize": 16,
            },
          ]
        }
      >
        Features and Feedback
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "minWidth": 35,
            "paddingRight": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 23,
              "width": 35,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "color": "#E1CD96",
              "fontFamily": "NexaBold",
              "paddingLeft": 4,
            },
          ]
        }
      >
        feedback@getmatch.co
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Nexa",
            "fontSize": 14,
          },
        ]
      }
    >
      We'd love to hear your feedbaack and ideas about our product and App.
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingBottom": 32,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#E1CD96",
          "borderBottomWidth": 1,
          "marginBottom": 2,
          "paddingBottom": 4,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "fontFamily": "SangBleuBP-LightItalic",
              "fontSize": 16,
            },
          ]
        }
      >
        Media Inquiries
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "opacity": 1,
          "paddingVertical": 4,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "minWidth": 35,
            "paddingRight": 4,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 23,
              "width": 35,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "color": "#E1CD96",
              "fontFamily": "NexaBold",
              "paddingLeft": 4,
            },
          ]
        }
      >
        pr@getmatch.co
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "Nexa",
            "fontSize": 14,
          },
          Object {
            "fontFamily": "NexaBold",
            "fontSize": 22,
          },
          Object {
            "color": "#E1CD96",
            "fontFamily": "SangBleuBP-LightItalic",
            "fontSize": 18,
          },
        ]
      }
    >
      Follow Us
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 16,
          "width": 202,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 38,
              "width": 38,
            }
          }
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 38,
              "width": 38,
            }
          }
        />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": 38,
              "width": 38,
            }
          }
        />
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
          "minHeight": 44,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontFamily": "Nexa",
              "fontSize": 14,
            },
            Object {
              "color": "#E1CD96",
            },
          ]
        }
      >
        Learn more at www.getmatch.co
      </Text>
    </View>
  </View>
</View>
`;
